src/assets/service-worker.ts

```diff
@@ -0,0 +1,31 @@
+/// <reference lib="webworker" />
+import { registerRoute } from 'workbox-routing';
+import { NetworkFirst } from 'workbox-strategies';
+import { CacheableResponsePlugin } from 'workbox-cacheable-response';
+import { ExpirationPlugin } from 'workbox-expiration';
+
+((self: ServiceWorkerGlobalScope) => {
```

Comment:
Maybe this can be simplified by [declaring `self`](https://www.devextent.com/create-service-worker-typescript/)?:

```ts
/// <reference lib="webworker" />
declare const self: ServiceWorkerGlobalScope;
```

Reply:
I looked at this with @decrek and there are two ways to do it:

```
declare const self: ServiceWorkerGlobalScope;
export default null // or export default {}
```
or do the self-invoking function to assign it to a new similarly named variable.

Without the export statement, it'll throw errors.

Reply:
🆗